uint16 center_x
uint16 center_y
uint16 x
uint16 y
uint16 width
uint16 height